### ISSUE TITLE
Shining Force 3 S2 Missing Water Effects and Bad Cursor

### ISSUE THREAD

**Issue by Lormaxx** [FCare/Kronos#933]
What's the full name of the game (including region) ?
Shining Force 3 Scenario 2 (J) with V21 English Translation Patch
What's your frontend ? (standalone, libretro, ...) ?
Libretro
What's your OS ? (windows, linux, OSX, ...) ?
Tried on Windows 10 1909 and Ubuntu 20.04.1 LTS (Focal Fossa) (dual boot)
What's your hardware ? (CPU, GPU, ...) ?
Ryzen 3400g + Vega 11 Latest Mesa Drivers on Linux. Latest Drivers on Windows 10
What's the storage type of the game (official cd, burned cd or cd image) ?
CDimage
If applicable, what's the format of your cd image ? (cue/bin, ccd/img, iso, chd, ...)
cue/bin
Are you using any setting different from default ? If so, precise

Tried various combinations of native resolution and 720p with GPU or CPU tessellation. Generally run 720p with GPU tessellation.
Did this game work previously ? Do you remember when ? Is it happening with other games ?
It has been fine up to this point with water effects. First time I have noticed the issue in Kronos 2.1.3. 

If applicable, provide error messages, logs, screenshots, or whatever additional information that could help debugging this

Beetle working
![Beetle Working](https://user-images.githubusercontent.com/69861748/96376440-7ac34d00-1144-11eb-9c70-d6075d401b88.png)

Kronos Not Working 
![Kronos Not Working](https://user-images.githubusercontent.com/69861748/96376446-857de200-1144-11eb-94ba-3d8b27f0d42c.png)

Save state loading right to the point the issue occurs
[Shining Force III - Scenario 2 - Nerawareta Miko State.zip](https://github.com/FCare/Kronos/files/5397953/Shining.Force.III.-.Scenario.2.-.Nerawareta.Miko.State.zip)

Memory save at the church right before the point it occurs. Leave town and skip through the cut scene

[Shining Force III - Scenario 2 - Nerawareta Miko Backup Save.zip](https://github.com/FCare/Kronos/files/5397955/Shining.Force.III.-.Scenario.2.-.Nerawareta.Miko.Backup.Save.zip)

Don't know if it will help but Devmiyax recently fixed an issue in Yaba Sanshiro 
https://github.com/Yabause/yabause/commit/eb751da4bcccb79275f4b1b74c7f67d922de452f

**Comment by Lormaxx** [comment]
Next Battle waterfall corrupt, select cursor and effects on the rocks

Beetle Working 

![Shining Force III - Scenario 2 - Nerawareta Miko-201018-154400](https://user-images.githubusercontent.com/69861748/96385668-047e1500-115b-11eb-9f88-36b9a2a504ca.png)

Kronos not working

![Shining Force III - Scenario 2 - Nerawareta Miko-201018-154504](https://user-images.githubusercontent.com/69861748/96385679-152e8b00-115b-11eb-9588-9d31e996b767.png)


Save and State files

[Shining Force III - Scenario 2 - Nerawareta Miko Save File.zip](https://github.com/FCare/Kronos/files/5398173/Shining.Force.III.-.Scenario.2.-.Nerawareta.Miko.Save.File.zip)
[Shining Force III - Scenario 2 - Nerawareta Miko State File.zip](https://github.com/FCare/Kronos/files/5398174/Shining.Force.III.-.Scenario.2.-.Nerawareta.Miko.State.File.zip)

**Comment by barbudreadmon** [comment]
Water is broken in 2.1.3 but should be ok in 2.1.4.
Here is the deal : i won't port 2.1.4 to the libretro port because it brings major regressions to other games like sega rally.
Until 2.1.4 is stabilized and ported to libretro, you can use the standalone version of kronos for SF3.

**Comment by Lormaxx** [comment]
Thanks Barbudreadmon for the info I'll try standalone when I get the chance... definitely understand about holding the Libretro build back due to the regressions as in all things emulation YMMV.

**Comment by barbudreadmon** [comment]
@Lormaxx the major regression in sega rally was fixed, so libretro is now in sync with standalone, can you confirm it fixes your water issue ?

**Comment by Lormaxx** [comment]
Yes I can confirm the issue is now resolved. I loaded the backup saves and both previous areas look good, as well as a third from my current progress. I should also note previous state files from 2.1.3 did not work loading them would produce a memory error in game. I even tried to load from back up memory and overwrite an existing state file, and encounter the same issue. However, once I removed all old states and created a fresh one on 2.1.4 it would load fine.

I'm under the assumption this was the result of the following commit: https://github.com/devmiyax/yabause/commit/fe4530e7a2b561e60052d756d898d01d7b34f375 and it's still a WIP. 

As the original issue is resolved I will mark this as closed. Thanks again,  barbudreadmon!
